domain/controllernode/service/service.go

```diff
@@ -81,6 +82,14 @@ type State interface {
 	// GetAPIAddressesForAgents returns the list of API addresses for the
 	// provided controller node that are available for agents.
 	GetAPIAddressesForAgents(ctx context.Context, ctrlID string) ([]string, error)
+
+	// SetPasswordHash sets the password hash for the given controller node.
+	SetPasswordHash(ctx context.Context, ctrlID string, passwordHash controllernode.ControllerPasswordHash) error
```

Comment:
These should be on the agent password service. If agent password is the wrong name for the service because it doesn't fit, then we should rename the service. The idea is that all password related things are in one central location.

Reply:
Yeah I would argue the controller is not an agent. The controller node exists in the controller database not model.

Happy to make that happen.

Reply:
Ah, ok leave it as it is.